server/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java

```diff
@@ -161,6 +161,17 @@ public boolean parsesArrayValue() {
         return false;
     }
 
+    /**
+     * Whether this mapper can handle an object value during document parsing. If true,
+     * when an object is encountered during parsing, the document parser will pass the
+     * whole object to the mapper. If false, the object name becomes part of the dotted
+     * field name of each internal value when subobjects property is set to false,
+     * otherwise it gets mapped to an ObjectMapper.
+     */
+    protected boolean parsesObject() {
+        return false;
+    }
```

Comment:
I double checked the mapper that you overrode this method for and it looks good, great work on figuring that out.

Reply:
It would be good to spend some time thinking how to test correctness for this value, and to make sure it does not go out of sync. We have `MapperTestCase` as a base class for all mappers tests, where we can add a new test method that all the subclasses will run. One idea would be to add a method that verifies that if the mapper supports parsing objects, it won't throw error when provided one (could be tricky because of the structure required, but perhaps we can leverage the existing `getSampleValueForDocument` which every test class already defines to provide valid values to be parsed. Also, we could verify (maybe easier) that when a mapper does not support parsing objects, it will throw an error when provided one (for which you can just try to parse a plain object and verify that parsing fails). 

Because this becomes possibly more involving, it would not be a bad idea to extract adding this method alone together with its related tests to a separate PR, even if the method is not yet used.